src/Workspaces/Core/Portable/Workspace/Solution/DocumentState.cs

```diff
@@ -19,6 +19,8 @@ namespace Microsoft.CodeAnalysis
 {
     internal partial class DocumentState : TextDocumentState
     {
+        private static readonly Func<string, PreservationMode, string> s_fullParseLog = (path, mode) => $"{path} : {mode}";
```

Comment:
Watch out for unintended allocations here. (Boxing of "mode" and the implicit string.Format).

```(path, mode) => path + " : " + mode.ToString();```

would be just as good and uses fewer allocations. (Although Enum.ToString isn't great).